app/controllers/sessions_controller.rb

```diff
@@ -2,18 +2,23 @@ class SessionsController < ApplicationController
   def new; end
 
   def create
-    user = User.find_by email: params[:session][:email].downcase
-    if user && user.authenticate(params[:session][:password])
-      log_in user
-      redirect_to user
+    @user = User.find_by email: params[:session][:email].downcase
```

Comment:
downcase và downcase! khác nhau gì em

Reply:
downcase return về giá trị không làm thay đổi object gọi nó
còn downcase! thì đổi luôn giá trị thằng object

Reply:
Nếu chuỗi không có kí tự nào in hoa thì sao em

Reply:
nếu chuỗi không có kí tự in hoa thì dùng downcase, giá trị trả về sẽ giống với chuỗi ban đầu.
Còn downcase! thì nó sẽ trả về nil, nhưng giá trị của chuỗi vẫn giữ nguyên.